shared/lsp/auth.ts

```diff
@@ -20,18 +20,18 @@ export async function getOrCreateAccessToken(
     name: string,
     note: string,
     api: API = new API()
-): Promise<string | undefined> {
+): Promise<string> {
     const accessToken = sourcegraph.configuration.get().get(name) as
         | string
         | undefined
     if (accessToken) {
         return accessToken
     }
     if (!accessTokenPromise) {
-        accessTokenPromise = (async (): Promise<string | undefined> => {
+        accessTokenPromise = (async (): Promise<string> => {
             const userId = await api.getUser()
             if (!userId) {
-                return undefined
+                return ''
```

Comment:
This seems weird - in TS, if something is typed as a `string` with a given semantic "access token" I would never expect it to be an empty string, because an empty string is not an access token. If the scenario is possible that there is no access token, that would be indicated and made type-checkable by making it a union `string | undefined` like before. If the thought behind this is to concatenate it into a string, I'd use `?? ''` at the place where it's concatenated

Reply:
I think this is especially important because we need to make sure the consumers don't create a URI like `https://@sourcegraph.com/...` but handle the case where it's not defined, to also omit the `@`